openwisp_notifications/static/openwisp-notifications/js/notifications.js

```diff
@@ -324,7 +348,41 @@ function notificationWidget($) {
         if (elem.hasClass('unread')) {
             markNotificationRead(elem.get(0));
         }
-        window.location = elem.data('location');
+
+        var notification = fetchedPages.flat().find((notification) => notification.id == elem.get(0).id.replace('ow-', ''));
+        if (notification.description) {
+            isNotificationDialogOpen = true;
+            const datetime = dateTimeStampToDateTimeLocaleString(new Date(notification.timestamp));
+            document.querySelector('.ow-dialog-notification-level-wrapper').innerHTML = `
+                        <div class="ow-notification-level-wrapper">
+                            <div class="ow-notify-${notification.level} icon"></div>
+                            <div class="ow-notification-level-text">${notification.level}</div>
+                        </div>
+                        <div class="ow-notification-date">${datetime}</div>
+            `;
+            document.querySelector('.ow-message-title').innerHTML = convertMessageWithRelativeURL(notification.message);
+            document.querySelector('.ow-message-description').innerHTML = notification.description;
+            $('.ow-dialog-overlay').removeClass('ow-hide');
+            if (notification.target_url) {
+                var target_url = new URL(notification.target_url, window.location.href);
+                $(document).on('click', '.ow-message-target-redirect', function () {
+                    window.location = target_url.pathname;
+                });
+                $('.ow-message-target-redirect').removeClass('ow-hide');
+            }
```

Comment:
I have this notification in my local environment 

```json
{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": "f012ed5e-673b-4158-9cf0-d7292b72d10e",
            "message": "<p>Something alarming happened!</p>",
            "description": "<p>Test.</p>",
            "unread": false,
            "target_url": "#",
            "email_subject": "[external.openwisp.test] Generic Notification Subject",
            "timestamp": "2024-05-24T20:12:41.217258+02:00",
            "level": "info"
        }
    ]
}
```

As you can see, there's not a valid target_url. I don't think it makes sense to reload the page after clicking on "Open" when the target_url is set to `#`.

Reply:
I will just make it disabled for `#` target_url